### DISCUSSION TITLE
create-starknet-app - Easy way to create a new frontend for your Starknet dApps

### DISCUSSION THREAD

**Discussion by walde-dev** [stark-rocket/.github#11] [Ideas]
Hey everyone!

I want to quickly introduce myself, before I tell you more about my idea:
- My name is Waldemar, I'm a frontend engineer from Munich, Germany
- I got into Starknet during EthMunich, where my team won a Starknet bounty, I met Lana there and she explained the whole concept behind zkProofs and the many use cases and advantages of building on Starknet, and even though I am quite a newbie when it comes to web3, I was really fascinated by it. I then participated in the Starknet DACH kickoff event in Berlin and also recently took part in the Starknet Hackerhouse in Istanbul.
- Now I am currently helping organize a Starknet meetup here in Munich.

It was in the Hackerhouse in Istanbul actually, where the idea first came to my mind, as I noticed that the biggest bottleneck for most project was the frontend, and not the smart contract part, or the backend. We have great libraries like `starknet-react` or `starknet.js` but there is still no single way to simply spin up a "starknet-ready" frontend, like you can with plain React (create-react-app) or Next.js (create-next-app).

As a frontend engineer I thought, that this is a missed opportunity and something that could really help many new Starknet developers. And thats how I got the idea behind create-starknet-app

It would basically be a CLI tool that streamlines the process of creating a starknet-ready frontend, while leaving room for other libraries that may or may not be needed for the apps of different developers. It would feature a React (or Next.js) boilerplate project, with a preconfigured and ready to use starknet.js / starknet-react integration, and some other things that may benefit the development of dApps. I'd really like to focus on a good DX and help people that might be less experienced in the frontend world, but rather prefer work on smart contracts or backend. 

Setup should be easy to follow and minimal, the idea is to just open your terminal, run something like `npx create-starknet-app@latest`, follow the step by step CLI guide, where you could e.g. pass in your contract address and have a project ready to go. 

I'm open for feedback, and especially hearing about the pain points you have experienced regarding setting up a frontend for your Starknet dApps. Thank you!

**Comment by YohanTz** [comment]
Hey @walde-dev! You should have a look at `create-starknet`: https://github.com/apibara/starknet-react/tree/main/packages/create-starknet, I made it months ago and I'm sure you can have some ideas to improve it. Let's discuss about what you had in mind :)

**Reply by walde-dev** [reply]
Hey! Thanks for reaching out, I didn't know something like this exists already, we can have a discussion about it for sure.
